aptos-move/framework/aptos-stdlib/sources/smart_tree.move

```diff
@@ -0,0 +1,896 @@
+module aptos_std::smart_tree {
+    use aptos_std::table_with_length::{Self, TableWithLength};
+    use std::option::{Self, Option};
+    use std::vector;
+    use std::debug;
+    use aptos_std::string;
+
+    const E_UNKNOWN: u64 = 0;
+    const E_TREE_NOT_EMPTY: u64 = 1;
+    const E_TREE_TOO_BIG: u64 = 2;
+    const E_END_ITERATOR: u64 = 3;
+
+    const NULL_INDEX: u64 = 0;
+
+    const DEFAULT_ORDER : u8 = 32;
+
+    struct Node has drop, store {
+        is_leaf: bool,
+        parent: u64,
+        children: vector<Child>,
+        prev: u64,
+        next: u64,
+    }
+
+    struct Child has copy, drop, store {
+        max_key: u64,
+        index: u64,
+    }
+
+    struct Iterator has copy, drop {
+        node_index: u64,
+        key_index: u64,
+        key: u64,
+    }
+
+    struct SmartTree<phantom V> has store {
+        root: u64,
+        nodes: TableWithLength<u64, Node>,
+        entries: TableWithLength<u64, V>,
```

Comment:
Yes, let me write more tests to make sure it works correctly with TableWithLength before I changing to the fancy bucket one, because I guess the bucket one may not be well tested.

Reply:
Done